### ISSUE TITLE
PIA Port Not applied to qbittorrent client

### ISSUE THREAD

**Issue by upcboy** [binhex/arch-qbittorrentvpn#26]
The new pia port does not get set on start up. The logs show PIA gives a port but it is not changed in the client. below is the relevant logs
```
qbitttorrent    | 2019-12-26 15:51:38,170 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Checking endpoint 'ca-toronto.privateinternetaccess.com' is port forward enabled...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:38,180 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Attempting to curl https://www.privateinternetaccess.com/vpninfo/servers?version=82...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:38,687 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Curl successful for https://www.privateinternetaccess.com/vpninfo/servers?version=82, response code 200
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:38,890 DEBG 'start-script' stdout output:
qbitttorrent    | [info] PIA endpoint 'ca-toronto.privateinternetaccess.com' is in the list of endpoints that support port forwarding
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:38,890 DEBG 'start-script' stdout output:
qbitttorrent    | [info] List of PIA endpoints that support port forwarding:-
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:38,891 DEBG 'start-script' stdout output:
qbitttorrent    | [info] ca-toronto.privateinternetaccess.com
qbitttorrent    | [info] ca-montreal.privateinternetaccess.com
qbitttorrent    | [info] ca-vancouver.privateinternetaccess.com
qbitttorrent    | [info] de-berlin.privateinternetaccess.com
qbitttorrent    | [info] de-frankfurt.privateinternetaccess.com
qbitttorrent    | [info] sweden.privateinternetaccess.com
qbitttorrent    | [info] swiss.privateinternetaccess.com
qbitttorrent    | [info] france.privateinternetaccess.com
qbitttorrent    | [info] czech.privateinternetaccess.com
qbitttorrent    | [info] spain.privateinternetaccess.com
qbitttorrent    | [info] ro.privateinternetaccess.com
qbitttorrent    | [info] israel.privateinternetaccess.com
qbitttorrent    | [info] Attempting to get dynamically assigned port...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:38,910 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Attempting to curl http://209.222.18.222:2000/?client_id=RED
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:39,839 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Curl successful for http://209.222.18.222:2000/?client_id=RED, response code 200
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:39,935 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Successfully assigned incoming port 56734
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:39,938 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Checking we can resolve name 'www.google.com' to address...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,035 DEBG 'start-script' stdout output:
qbitttorrent    | [info] DNS operational, we can resolve name 'www.google.com' to address '172.217.1.4'
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,039 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Attempting to get external IP using Name Server 'ns1.google.com'...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,248 DEBG 'start-script' stdout output:
qbitttorrent    | [info] Successfully retrieved external IP address 172.98.67.4
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,293 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] qBittorrent listening interface IP 0.0.0.0 and VPN provider IP 10.23.10.10 different, marking for reconfigure
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,301 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] qBittorrent not running
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,310 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] Privoxy not running
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,311 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] qBittorrent incoming port 6881 and VPN incoming port 56734 different, marking for reconfigure
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,311 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] Removing session lock file (if it exists)...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,315 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] Attempting to start qBittorrent...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,343 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] qBittorrent process started
qbitttorrent    | [info] Waiting for qBittorrent process to start listening on port 8088...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,470 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] qBittorrent process listening on port 8088
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:40,507 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] Attempting to start Privoxy...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:41,513 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] Privoxy process started
qbitttorrent    | [info] Waiting for Privoxy process to start listening on port 8118...
qbitttorrent    |
qbitttorrent    | 2019-12-26 15:51:41,521 DEBG 'watchdog-script' stdout output:
qbitttorrent    | [info] Privoxy process listening on port 8118
```

**Comment by BigC27** [comment]
Just wanted to add that I'm also having this issue. 
Logs show the same sequence as above.

**Comment by binhex** [comment]
@upcboy if you follow the logs later on your should see qbittorrent incoming port gets set, and if you login to web ui and click on options and click on 'connection' tab you will see the port is set. 

ive just tested this and it is being set correctly for me.

**Comment by upcboy** [comment]
@binhex I've been doing that as a workaround but I assumed since it was asking for a port it would be setting the port automatically on startup. This should be as simple as doing 

`sed -i '/Connection\\PortRangeMin=/c\Connection\\PortRangeMin=PIA_Port' /qBittorrent/conf/qBittorrent.conf`

**Comment by binhex** [comment]
i think you misunderstand, im not suggesting you manually set the port, im saying it should be (and is for me) getting automatically set, check the web ui, the port should match the pia incoming port shown in the log.

**Comment by upcboy** [comment]
I just restarted the container and the logs show below for the port 
```
2020-01-03 21:49:58,854 DEBG 'start-script' stdout output:
[info] Successfully assigned incoming port 30469
```

but my current port is set to 24995 in the webUI when I check (which was the last port i manually find from the logs.)

**Comment by binhex** [comment]
Sorry guys my mistake I was using a test tagged version I have now replicated the problem and identified the fix look out for a new version soon

**Comment by binhex** [comment]
now fixed, please pull down latest image.

**Comment by upcboy** [comment]
I'm still seeing the same behavior and I have made sure I have the latest build on docker hub.

**Comment by binhex** [comment]
hmm just double checked and its def working for me, so the incoming port shown in the supervisord.log file is not getting set in qbittorrent? (web ui/options/connection tab/port used for incoming connections)

**Comment by upcboy** [comment]
my logs show 

```
[info] Successfully assigned incoming port 28671
```
but the webui shows 47417
![image](https://user-images.githubusercontent.com/5158170/71935099-9805cc80-316b-11ea-926a-e269905da078.png)

**Comment by binhex** [comment]
are you 100% sure you have got the latest image pulled down?, issue the following:-

docker pull binhex/arch-qbittorrentvpn:latest

then delete your existing container and re-create it.

**Comment by upcboy** [comment]
The digest of my images matches the current lates from docker hub
```
REPOSITORY                     TAG                 DIGEST                                                                    IMAGE ID            CREATED             SIZE
binhex/arch-qbittorrentvpn     latest              sha256:be7780965e945ace78e66480b176448d4e45e360c3bd2678451f0a26f9085124   2c0eb5f10c57        2 days ago          1.02GB
```

**Comment by upcboy** [comment]
I found the issue. It looks like in your commit for the fix you hard coded the webui port to 2100 instead of using the variable WEBUI_PORT you had been using before. This would explain why it works for you but not for me since my webui port does not match yours. I can run that same curl command and change my port with out issue if i change the port to my correct webui port.

**Comment by binhex** [comment]
ugghh!, thanks for that, must of hard coded it during testing!, merged and image built, please pull down and give it a whirl.

**Comment by BigC27** [comment]
Yep working for me now. Cheers!

**Comment by upcboy** [comment]
Fixed by 7151044024f92d154ccdb055c331c6f10d37038b
